core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -197,7 +211,11 @@ public final T nameResolverFactory(NameResolver.Factory resolverFactory) {
     Preconditions.checkState(directServerAddress == null,
         "directServerAddress is set (%s), which forbids the use of NameResolverFactory",
         directServerAddress);
-    this.nameResolverFactory = resolverFactory;
+    if (resolverFactory != null) {
```

Comment:
I think these experimental APIs are rarely used, and the javadoc didn't say what's the behavior for null argument, so if we change the behavior now, there shouldn't be impact.